Q1.cpp

```diff
@@ -0,0 +1,31 @@
+#include <vector>
+#include <string>
+
+// when both strings are empty, they are considered permutations
+bool is_permutation(const std::string& str1, const std::string& str2){
+	bool out = false;
+	
+	if(str1.size()==str2.size()){
+		
+		std::vector<bool> check(str1.size(),true); // check this initialization
```

Comment:
Initialisation is fine, but a better name for the variable won't hurt :) 'available', 'not_used' ?